kafka-0.10.x/src/main/scala/monix/kafka/KafkaConsumerObservable.scala

```diff
@@ -62,44 +65,61 @@ trait KafkaConsumerObservable[K, V, Out] extends Observable[Out] {
   private def feedTask(out: Subscriber[Out]): Task[Unit] = {
     Task.create { (scheduler, cb) =>
       implicit val s = scheduler
-      val feedTask = consumer.flatMap { c =>
-        // Skipping all available messages on all partitions
-        if (config.observableSeekOnStart.isSeekEnd) c.seekToEnd(Nil.asJavaCollection)
-        else if (config.observableSeekOnStart.isSeekBeginning) c.seekToBeginning(Nil.asJavaCollection)
-        // A task to execute on both cancellation and normal termination
-        val onCancel = cancelTask(c)
-        runLoop(c, out).guarantee(onCancel)
-      }
-      feedTask.runAsync(cb)
+      val startConsuming =
+        Resource
+          .make(consumer) { c =>
+            // Forced asynchronous boundary
+            Task.evalAsync(consumer.synchronized(blocking(c.close()))).memoizeOnSuccess
+          }
+          .use { c =>
+            // Skipping all available messages on all partitions
+            if (config.observableSeekOnStart.isSeekEnd) c.seekToEnd(Nil.asJavaCollection)
+            else if (config.observableSeekOnStart.isSeekBeginning) c.seekToBeginning(Nil.asJavaCollection)
+            // A task to execute on both cancellation and normal termination
+            pollHeartbeat(c)
+              // If polling fails the error is reported to the subscriber and
+              // wait 1sec as a rule of thumb leaving enough time for the consumer
+              // to recover and reassign partitions
+              .onErrorHandleWith(ex => Task(out.onError(ex)))
+              .loopForever
+              .startAndForget
```

Comment:
It might make sense to do `Task.race(runLoop(c, out), pollHearbeat(c).loopForever)` to stop the `Task` in case `runLoop` stops. 

`isAcked` should be `true` so it's not a big deal but it's safer